pkg/asset/installconfig/powervs/session.go

```diff
@@ -182,6 +185,67 @@ func (c *BxClient) ValidateAccountPermissions() error {
 	return nil
 }
 
+//ValidateDhcpService checks for existing dhcp instance for the provided PowerVS cloud instance
+func (c *BxClient) ValidateDhcpService(ctx context.Context, svcInsID string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, helpers.PIGetTimeOut)
+	defer cancel()
+	res := make(chan int)
+
+	go func() error {
+		//create Power VS DHCP Client
+		dhcpClient := instance.NewIBMPIDhcpClient(ctxWithTimeout, c.PISession, svcInsID)
+		//Get all DHCP Services
+		dhcpServers, err := dhcpClient.GetAll()
+		if err != nil {
+			return err
+		}
+		res <- len(dhcpServers)
+		close(res)
+		return nil
+	}()
+	for {
+		select {
+		case noOfDhcpServices := <-res:
+			if noOfDhcpServices > 0 {
+				return fmt.Errorf("DHCP service already exists for provided cloud instance")
+			}
+			return nil
+		case <-ctxWithTimeout.Done():
+			return ctxWithTimeout.Err()
+		}
+	}
+}
+
+//ValidateCloudConnectionInPowerVSRegion counts cloud connection in PowerVS Region
+func (c *BxClient) ValidateCloudConnectionInPowerVSRegion(ctx context.Context, svcInsID string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, helpers.PIGetTimeOut)
```

Comment:
Generally the child context takes on the name ctx also. More of nit picks but it is consistent in our code. Also can look into using a specific time like 60*time.Second or whatever the equivalent is for PIGetTimeOut